src/Symfony/Component/Finder/Iterator/VcsIgnoredFilterIterator.php

```diff
@@ -36,23 +36,35 @@ public function accept(): bool
     {
         $file = $this->current();
 
-        $fileRealPath = $file->getRealPath();
+        $fileRealPath = $this->normalizePath($file->getRealPath());
         if ($file->isDir()) {
             $fileRealPath .= '/';
         }
 
         $parentDirectory = $fileRealPath;
 
-        do {
-            $parentDirectory = \dirname($parentDirectory);
+        while (true) {
+            $newParentDirectory = \dirname($parentDirectory);
+
+            // dirname('/') = '/'
+            if ($newParentDirectory === $parentDirectory) {
+                break;
+            }
+
+            $parentDirectory = $newParentDirectory;
+
+            if (0 !== strpos($parentDirectory, $this->baseDir)) {
+                break;
+            }
+
```

Comment:
```suggestion
        $prevDirectory = $parentDirectory;
        while (($parentDirectory = \dirname($parentDirectory)) !== $prevDirectory && 0 === strpos($parentDirectory, $this->baseDir)) {
            $prevDirectory = $parentDirectory;

```

Reply:
This is less legible IMO.

Reply:
at least the first condition should be kept in the while loop